kingpin/actors/rightscale/test/test_api.py

```diff
@@ -116,7 +116,7 @@ def test_find_by_name_and_keys(self):
         ret = yield self.client.find_by_name_and_keys(
             collection=mock.MagicMock(), exact=True,
             name='FakeResource', href='/123')
-        self.assertEquals(ret, None)
```

Comment:
:-/ Not what I meant. If `exact=True` is provided that means it should be exactly 1 object, right? So it probably should return an object or raise an exception. 

If `exact=False` then it's a search, and therefore always a list.